### PR TITLE
Fixes custom Brave ad notifications should not be shown on Dev channel builds - 1.20.x

### PR DESCRIPTION
**Pull request by tmancey**
Uplift of https://github.com/brave/brave-core/pull/7608
Resolves https://github.com/brave/brave-browser/issues/13592

- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR.
- [x] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by kjozwiak** [comment]
Restarting `macOS` due to `build` failing as per https://ci.brave.com/job/pr-brave-browser-issues-13592-1.20.x-macos/1/execution/node/155/log/. However, this is a known issue that was fixed via https://github.com/brave/brave-core/pull/7599. Going to attempt one more restart and if `macOS` fails again due to the same failure, we'll just merge as it's a known problem and has been fixed.

```
05:31:23  FAILED: gen/chrome/browser/resources/settings/settings.rollup.js gen/chrome/browser/resources/settings/lazy_load.rollup.js gen/chrome/browser/resources/settings/shared.rollup.js gen/chrome/browser/resources/settings/build_manifest.json
```

**Comment by tmancey** [comment]
Restarting macOS due to same reason as posted by @kjozwiak

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.20.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/7608#issuecomment-763162251. As per discussions via Slack, this technically can't be checked till it lands in `1.20.x` but we did make sure that the PR didn't regress Nightly.

**Pull request merged**
